### PR TITLE
Datastore rules services

### PR DESCRIPTION
**Pull request by mpkocher**

### CODE REVIEW

**Comment by mpkocher** [comment]
@natechols can you take a look at this?

File: smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/jobs/JobModels.scala [hunk 1]
```diff
@@ -291,4 +291,8 @@ object JobModels {
       description: String,
       taskOptions: Seq[PipelineOptionViewRule])
 
+  // FIXME(mkocher)(2016-8-18) All of these View rules should probable be migrated to a central location
+  case class DataStoreFileViewRule(sourceId: String, name: Option[String], description: Option[String], isHidden: Boolean = false, fileTypeId: String)
```
Comment (natechols) [inline smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/jobs/JobModels.scala]: shouldn't the required argument (fileTypeId) appear before an optional one (isHidden)?

File: smrt-analysis/src/test/resources/pipeline-datastore-view-rules/pipeline_datastore_view_rules-dev-01.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      "sourceId": "pbsmrtpipe::pbsmrtpipe.log",
+      "name": "Name",
+      "description": "Description",
+      "isHidden": false
+    },
+    {
+      "fileTypeId": "PacBio.FileTypes.log",
+      "sourceId": "pbsmrtpipe::master.log",
+      "isHidden": true,
+      "description": "",
```
Comment (natechols) [inline smrt-analysis/src/test/resources/pipeline-datastore-view-rules/pipeline_datastore_view_rules-dev-01.json]: I think this needs to be null to cover the None case.

Comment (mpkocher) [inline smrt-analysis/src/test/resources/pipeline-datastore-view-rules/pipeline_datastore_view_rules-dev-01.json]: I was getting some serialization errors when I initially had null in there. I've updated the test data. Hopefully this was a local caching issue.

**Comment by mpkocher** [comment]
@natechols Can you merge if this looks good to you?

**Pull request merged**